pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -171,6 +171,21 @@ def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', ve
 
         self.kth_value, self.mode = mode_dict[mode]
 
+    def _validate_condition_metric(self, logs):
+        monitor_val = logs.get(self.monitor)
+
+        if monitor_val is None:  # self.monitor wasn't found in the metrics dict
+            if self.verbose > 0:
+                error_msg = (f'Checkpointing conditioned on metric `{self.monitor}`'
+                             f' which is not available. Either add `{self.monitor}` to the return of'
+                             f' validation_epoch end or modify your CheckPoint callback to use any of the'
+                             f' following: `{"`, `".join(list(logs.keys()))}`')
+                rank_zero_warn(error_msg, RuntimeWarning)
+
+            return False
+
+        return True
```

Comment:
this code can be improved. 
my suggestion:
```python 
def _validate_condition_metric(self, logs):
    monitor_val = logs.get(self.monitor)
    found = monitor_val is not None
    if not found and self.verbose > 0:
        error_msg = (
            f'Checkpointing conditioned on metric `{self.monitor}`'
            f' which is not available. Either add `{self.monitor}` to the return of'
            f' validation_epoch end or modify your CheckPoint callback to use any of the'
            f' following: `{"`, `".join(list(logs.keys()))}`'
        )
        rank_zero_warn(error_msg, RuntimeWarning)
    return found
```
1. removed unnecessary nesting of if
2. by introducing a new variable with a good name, we immediately eliminate the need to explain the code in a comment
3. there is a single return statement

Reply:
my concern is why the return state is needed if it not used anywhere...

Reply:
oh yeah,, not required.

Reply:
I think it is because the PR author copied the code from the early stop callback
but there the return is also not used

Reply:
I think nonetheless, the code is much more readable this way, if we have the return or not

Reply:
yeah @awaelchli, will make the changes.